Unknown pragma warning when using -DENABLE_STATIC_LIBS=ON

src/parseAPI/src/Parser.C:297: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
      297 | #pragma omp parallel for schedule(dynamic)
          | 
    src/parseAPI/src/Parser.C:619: error: ignoring #pragma omp task [-Werror=unknown-pragmas]
      619 | #pragma omp task firstprivate(frame, recursive)
          | 
    src/parseAPI/src/Parser.C:656: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
      656 | #pragma omp parallel shared(work_queue)
          | 
    src/parseAPI/src/Parser.C:658: error: ignoring #pragma omp master [-Werror=unknown-pragmas]
      658 | #pragma omp master
          | 
    src/parseAPI/src/Parser.C:784: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
      784 | #pragma omp parallel for schedule(dynamic)
          | 
    src/parseAPI/src/Parser.C:1069: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
     1069 | #pragma omp parallel for schedule(dynamic)
          | 
    src/parseAPI/src/Parser.C:1169: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
     1169 | #pragma omp parallel for schedule(dynamic)
          | 
    src/symtabAPI/src/Symtab.C:642: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
      642 |     #pragma omp parallel for schedule(dynamic)
          | 
    src/symtabAPI/src/Symtab.C:661: error: ignoring #pragma omp parallel [-Werror=unknown-pragmas]
      661 |   #pragma omp parallel for
          | 
    cc1plus: all warnings being treated as errors
    make[2]: *** [parseAPI/CMakeFiles/parseAPI_static.dir/build.make:76: parseAPI/CMakeFiles/parseAPI_static.dir/src/Parser.C.o] Error 1
    make[2]: *** Waiting for unfinished jobs....
    cc1plus: all warnings being treated as errors
    make[2]: *** [symtabAPI/CMakeFiles/symtabAPI_static.dir/build.make:141: symtabAPI/CMakeFiles/symtabAPI_static.dir/src/Symtab.C.o] Error 1
    make[2]: *** Waiting for unfinished jobs....
    make[1]: *** [CMakeFiles/Makefile2:1189: parseAPI/CMakeFiles/parseAPI_static.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    make[1]: *** [CMakeFiles/Makefile2:1133: symtabAPI/CMakeFiles/symtabAPI_static.dir/all] Error 2
    make: *** [Makefile:130: all] Error 2

This is weird because USE_OpenMP=ON during this build. When USE_OpenMP=OFF, you get these warnings as expected.

--- a/cmake/warnings.cmake
+++ b/cmake/warnings.cmake
@@ -96,6 +96,11 @@ endif()
 if(DYNINST_WARNINGS_AS_ERRORS)
     list(APPEND REQUESTED_WARNING_FLAGS "Werror")
     message(STATUS "DYNINST_WARNINGS_AS_ERRORS set: treating warnings as errors")
+    
+    # If not building with OpenMP or if static libs are enabled, ignore OpenMP pragma warnings
+    if(NOT USE_OpenMP OR ENABLE_STATIC_LIBS)
+    	list(APPEND REQUESTED_WARNING_FLAGS "Wno-unknown-pragmas")
+    endif()
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang)$")

